### PR TITLE
[New Concept] AI Foundation Models

### PR DESCRIPTION
**Pull request by yangc95**
### Description

As discussed here: https://github.com/Codecademy/docs/pull/2165#issuecomment-1569038020
- Create new concept "Foundation Models"

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by yangc95** [comment]
@Hassan-fayez Would you like to take a look at this entry as well (one if you're comfortable)? Please let me know otherwise. Foundation model is the term we decided to go with for this concept entry so it would be a good idea to change the file path for the term entry to the correct one. Thanks!

**Review by HishamT — CHANGES_REQUESTED** [review]
Hey @yangc95, can you add some examples of AI foundation models? :grinning: 

GPT, VAE, BERT, etc...

Thanks!

**Comment by yangc95** [comment]
@HishamT Added those examples!

**Review by HishamT — APPROVED** [review]
LGTM!

**Review by HishamT — APPROVED** [review]
LGTM!!

**Comment by yangc95** [comment]
@HishamT I added some additional info on foundation models based on this source: https://crfm.stanford.edu/assets/report.pdf#technology

File: content/ai/concepts/foundation-models/foundation-models.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+## Model Architecture
+
+Foundation models use various architecture types. One of which includes the transformer architecture (like evolved transformer or vanilla transformer).
+
+## Training
+
+Foundation models are based on [deep learning](https://www.codecademy.com/resources/docs/ai/machine-learning) standards and transfer learning.
+
+Pre-training models on data is vital in transfer learning, which deals with the application of knowledge learned from one task onto another task. As computer hardware increase in speed and memory, the amount of training data that is able to be processed increases as well, making foundation models possible.
```
Comment (SSwiniarski) [inline content/ai/concepts/foundation-models/foundation-models.md:43]:
```suggestion
Pre-training models on data is vital in transfer learning, which deals with the application of knowledge learned from one task onto another task. As computer hardware increases in speed and memory, the amount of training data that is able to be processed increases as well, making foundation models possible.
```

File: content/ai/concepts/foundation-models/foundation-models.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+## Existing Models
+
+Some examples of existing foundation models include:
+
+- BERT (Bidirectional Encoder Representations from Transformers) is a model introduced by Google in 2018.
+- LaMDA (Language Model for Dialogue Applications) is a model introduced by Google in 2020, and powers Bard which was released in March 2023.
+- GPT (Generative Pre-trained Transformer)(https://www.codecademy.com/resources/docs/ai/chatgpt) is a model that was first introduced by OpenAI in 2018. It is trained on text and code, and powers [ChatGPT](https://www.codecademy.com/resources/docs/ai/chatgpt) which was released in November 2022.
```
Comment (SSwiniarski) [inline content/ai/concepts/foundation-models/foundation-models.md:51]:
```suggestion
- GPT (Generative Pre-trained Transformer) is a model that was first introduced by OpenAI in 2018. It is trained on text and code, and powers [ChatGPT](https://www.codecademy.com/resources/docs/ai/chatgpt) which was released in November 2022.
```

File: content/ai/concepts/foundation-models/foundation-models.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+Tags:
+  - 'AI'
+  - 'NLP'
+  - 'Deep Learning'
+CatalogContent:
+  - 'paths/data-science-nlp'
+  - 'intro-to-chatgpt'
+---
+
+**Foundation models** power AI systems that are trained on text, images, code, and other formats of data. These models pre-train on large amounts of unlabeled data from the internet. They can then be fine-tuned to perform various downstream tasks, like generating content(referred to as [GenAI](https://www.codecademy.com/resources/docs/ai/generative-ai)).
```
Comment (SSwiniarski) [inline content/ai/concepts/foundation-models/foundation-models.md:15]:
```suggestion
**Foundation models** power AI systems that are trained on text, images, code, and other formats of data. These models pre-train on large amounts of unlabeled data from the internet. They can then be fine-tuned to perform various downstream tasks, like generating content(referred to as [generative AI (GenAI)](https://www.codecademy.com/resources/docs/ai/generative-ai)).
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@yangc95 I did a second review and only had a few minor comments.

**Comment by yangc95** [comment]
Thanks @SSwiniarski added those suggestions!

Also @HishamT did you want to take another look after I added additional context.

**Review by SSwiniarski — APPROVED** [review]
@yangc95 Looks good!

**Comment by SSwiniarski** [comment]
@HishamT are we good to merge this?

**Pull request merged**